controllers/gitrepository_controller.go

```diff
@@ -428,6 +429,13 @@ func (r *GitRepositoryReconciler) reconcileStorage(ctx context.Context,
 // change, it short-circuits the whole reconciliation with an early return.
 func (r *GitRepositoryReconciler) reconcileSource(ctx context.Context,
 	obj *sourcev1.GitRepository, commit *git.Commit, includes *artifactSet, dir string) (sreconcile.Result, error) {
+	// Exit early, if we need to use libgit2 AND managed transport hasn't been intialized.
+	if !managed.Enabled() && obj.Spec.GitImplementation == sourcev1.LibGit2Implementation {
+		fmt.Println(managed.Enabled())
+		return sreconcile.ResultEmpty, serror.NewStalling(
+			errors.New("libgit2 managed transport not initialized"), "Libgit2TransportNotEnabled",
+		)
+	}
```

Comment:
I think we need to test this case of `reconcileSource()` to ensure we don't change it unintentionally in the future. But to test this, we need to run the test before initializing managed transport, which won't be easy or elegant based on what we've done in the libgit2 package before, many workarounds.
I think it'd be better to do something similar to `HelmRepositoryOCIReconciler` and `HelmChartReconciler` where they accept a function type field `RegistryClientGeneratorFunc` for generating registry client, which can be easily used for mocking. In this case, we can inject a mock of transport initialized check to be able to test when it fails.
So, let's add a new field in `GitRepositoryReconciler`, which can be set to `managed.Enabled()` in main.go, but can be overridden in the tests where we want it to be something else. And add a new `reconcileSource` test for this, where we make sure the conditions are set properly and returns the appropriate error.